app/services/spotlight/clone_translated_page_from_locale.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Spotlight
+  ##
+  # A service class to encapsulate business logic
+  # around cloning and destroying translated pages
+  class CloneTranslatedPageFromLocale
+    attr_reader :locale, :page
+
+    def initialize(locale:, page:)
+      @locale = locale
+      @page = page
+    end
+
+    def self.call(locale:, page:)
+      new(locale: locale, page: page).clone
+    end
+
+    def clone
```

Comment:
`clone` strikes me as a deceptive name if it will destroy an existing object, but I don't have a better suggestion off-hand.